test/parallel/test-stream-reduce.js

```diff
@@ -0,0 +1,112 @@
+'use strict';
+
+const common = require('../common');
+const {
+  Readable,
+} = require('stream');
+const assert = require('assert');
+
+function sum(p, c) {
+  return p + c;
+}
+
+{
+  // Does the same thing as `(await stream.toArray()).reduce(...)`
+  (async () => {
+    const tests = [
+      [[], sum, 0],
+      [[1], sum, 0],
+      [[1, 2, 3, 4, 5], sum, 0],
+      [Array(100).fill().map((_, i) => i), sum, 0],
```

Comment:
```suggestion
      [Array(100).keys(), sum, 0],
```

Reply:
Ooh that's clever and my new favorite way to get a range in JavaScript.